packages/twenty-server/src/modules/workflow/common/workspace-services/workflow-version-step.workspace-service.ts

```diff
@@ -0,0 +1,361 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { join } from 'path';
+
+import { Repository } from 'typeorm';
+import { v4 } from 'uuid';
+
+import { TwentyORMManager } from 'src/engine/twenty-orm/twenty-orm.manager';
+import { WorkflowVersionWorkspaceEntity } from 'src/modules/workflow/common/standard-objects/workflow-version.workspace-entity';
+import {
+  WorkflowAction,
+  WorkflowActionType,
+} from 'src/modules/workflow/workflow-executor/workflow-actions/types/workflow-action.type';
+import { isDefined } from 'src/utils/is-defined';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { WorkflowBuilderWorkspaceService } from 'src/modules/workflow/workflow-builder/workflow-builder.workspace-service';
+import { ServerlessFunctionService } from 'src/engine/metadata-modules/serverless-function/serverless-function.service';
+import { WorkflowRecordCRUDType } from 'src/modules/workflow/workflow-executor/workflow-actions/record-crud/types/workflow-record-crud-action-input.type';
+import { WorkflowActionDTO } from 'src/engine/core-modules/workflow/dtos/workflow-step.dto';
+import { INDEX_FILE_NAME } from 'src/engine/core-modules/serverless/drivers/constants/index-file-name';
+import { CodeIntrospectionService } from 'src/modules/code-introspection/code-introspection.service';
+import {
+  WorkflowVersionStepException,
+  WorkflowVersionStepExceptionCode,
+} from 'src/modules/workflow/common/exceptions/workflow-version-step.exception';
+
+const TRIGGER_STEP_ID = 'trigger';
+
+const BASE_STEP_DEFINITION = {
+  outputSchema: {},
+  errorHandlingOptions: {
+    continueOnFailure: {
+      value: false,
+    },
+    retryOnFailure: {
+      value: false,
+    },
+  },
+};
+
+@Injectable()
+export class WorkflowVersionStepWorkspaceService {
+  constructor(
+    private readonly twentyORMManager: TwentyORMManager,
+    private readonly workflowBuilderWorkspaceService: WorkflowBuilderWorkspaceService,
+    private readonly serverlessFunctionService: ServerlessFunctionService,
+    private readonly codeIntrospectionService: CodeIntrospectionService,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+  ) {}
+
+  private async getStepDefaultDefinition({
+    type,
+    workspaceId,
+  }: {
+    type: WorkflowActionType;
+    workspaceId: string;
+  }): Promise<WorkflowAction> {
+    const newStepId = v4();
+
+    switch (`${type}`) {
```

Comment:
weird, why do you need this?

Reply:
that is because of RECORD_CRUD.CREATE types, think we should remove those and have CREATE_RECORD, UPDATE_RECORD, etc... types